chapters/1/2-problemy.tex

```diff
@@ -0,0 +1,7 @@
+\newline \indent- W związku z zakupem serwera posiadającego większą moc obliczeniową, system STOS zostanie przeniesiony z obecnego komputera na nową maszynę. Tworzone rozwiązanie musi być dostosowane do systemu gospodarza oraz powinien jak najlepiej wykorzystywać jego zasoby.
+\newline \indent Przygotowanie platformy STOS polega na ręcznym uruchomieniu skryptów, które tworzą kontener z modułem kompilującym oraz modułu sprawdzającego. W przypadku zatrzymania działania systemu, awarii, konieczności zmiany wersji, odłączenia zasilania, lub zawieszenia się procesu, należy podłączyć się do maszyny i uruchomić lub zrestartować wcześniej wspomniane skrypty. Jest to uciążliwe ze względu na konieczność regularnego monitorowania stanu platformy i konieczności posiadania wiedzy o wymaganej konfiguracji oraz kolejności uruchamiania. Docelowe rozwiązanie, powinno umożliwić na ciągłe działanie systemu, szczególnie podatnych na awarię modułów kompilujących. Cel można osiągnąć poprzez wysyłanie wiadomości potwierdzających działanie komponentu (tak zwane wiadomości typu „healthcheck”), równoległe działanie kilku instancji serwisów oraz monitorowanie sygnałów zakończenia procesów i próbę ich naprawy lub wznowienia. W naszym systemie zdecydowaliśmy się na zapewnienie automatycznego wznawiania jednocześnie działających instancji modułu kompilującego do określonej liczby.
+\newline \indent Kluczową funkcjonalnością systemu STOS, jest ocena pracy studenta. Ocena bierze pod wzgląd pomyślność kompilacji kodu programu, identyczność wyniku z oczekiwanym rezultatem oraz szybkość wykonania obliczeń. Wykonując serię ocen tego samego rozwiązania, można zauważyć nieidempotentność oceny czasu wykonania algorytmu. Jest ona silnie uzależniona od aktualnego obciążenia serwera gospodarza systemu STOS. W przypadku wystąpienia procesu działającego w tle używającego dużej ilości zasobów obliczeniowych oceniane w tym momencie rozwiązanie studenta może wykonywać się dłużej, co może skutkować niezaliczeniem testu z powodu przekroczenia czasu. Problem można rozwiązać poprzez wydzielenie stałego rozmiaru zasoby obliczeniowe procesora i ilości pamięci dla modułu kompilującego. Jest to nieefektywne rozwiązanie, ponieważ część zasobów może być niewykorzystana, a przeznaczenie zbyt małej ich ilości może spowodować błędy podczas uruchomienia rozwiązania studenta. Alternatywą, która nie posiada tego problemu i gwarantuje uniezależnienie czasu wykonania od obciążenia systemu, jest zliczanie wykonanych niskopoziomowych operacji. Dla danego algorytmu liczba operacji, które wykona, jest stała, dla danego zestawu danych wejściowych. Taka informacja gwarantuje sprawiedliwą ocenę programu pod względem szybkości wykonania. Ze względu na złożoność w implementacji oraz priorytetyzację pozostałych problemów, mechanizm nie zostanie zaimplementowany w ramach tej pracy.
```

Comment:
To jest to samo co u Adama

Reply:
U mnie jest trochę bardziej implementacyjnie i moim zdaniem może być wzmianka o tym zarówno we wstępie nakreślającym problemy systemu jak i w opisie komponentów.